### DISCUSSION TITLE
Unable to extract PDF content problem

### DISCUSSION THREAD

**Discussion by paksak** [dadoonet/fscrawler#1548] [Q&A] [answered]
Hi, David. Another question again.

I installed FS Crawler 2.10-SNAPSHOT and tesseract 5.0 on CentOS-7 with Leptonica-1.79, devtools 7 
I uploaded pdf file and then error message like below appeared.

#############################################################################################
18:25:06,791 WARN  [f.p.e.c.f.t.TikaDocParser] Failed to extract [100000] characters of text for [/usr/local/lib/apache-tomcat-8.5.83/webapps/ROOT/contract/uzz.pdf]
: Unable to extract PDF content -> Unable to end a page -> I regret that I couldn't find an OCR parser to handle image/ocr-png.Please set the OCR_STRATEGY to NO_OCR or configure yourOCR parser correctly
#############################################################################################

The file is uploaded but it doesn't have its text contents. Just searched by title.
I installed fscrawler and tesseract 4.0 on Ubuntu before and it worked well, so I could read pdf contents on elastic.
But with this combination, some problem occured.
What did I do wrong ?

##############################################################
_setting.yml
###############################################################
name: "contract"
fs:
  url: "/usr/local/lib/apache-tomcat-8.5.83/webapps/ROOT/contract"
  update_rate: "15m"
  excludes:
  - "*/~*"
  json_support: false
  filename_as_id: false
  add_filesize: true
  remove_deleted: true
  add_as_inner_object: false
  store_source: false
  index_content: true
  attributes_support: false
  raw_metadata: false
  xml_support: false
  index_folders: true
  lang_detect: false
  continue_on_error: false
  ocr:
    path: "/usr/local/tesseract-5.0/bin"
    data_path: "/usr/local/tesseract-5.0/share/tessdata/"
    language: "eng"
    enabled: true
    pdf_strategy: "ocr_and_text"
  follow_symlinks: false
elasticsearch:
  nodes:
  - url: "http://192.168.150.140:9200"
  bulk_size: 100
  flush_interval: "5s"
  byte_size: "10mb"
  username: "elastic"
  password: "*************"
  ssl_verification: false

**Comment by paksak** [comment, marked as answer]
I found the reason.
In tesseract 5.0, libpng12-devel rpm pkg was needed ( libpng-devel was used for version 4.0)
After 'yum install libpng12-devel', it works well.
thank you everyone always

What I did is like below.

################################################
first , I installed related rpm packages
######################################################
yum install -y autoconf automake libtool pkgconfig.x86_64
yum install -y libpng12-devel.x86_64 libjpeg-devel libtiff-devel.x86_64 zlib-devel.x86_64

#####################################################
Secon, updated gcc version
#####################################################
yum install centos-release-scl
yum install devtoolset-9-toolchain
scl enable devtoolset-9 bash

After these, Non-binary pdf file text contents was uploaded and could be read.

**Reply by dadoonet** [reply]
Amazing! Could you update the documentation?

**Reply by paksak** [reply]
Sure. I will do soon
